File: app/Comment.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Comment extends Model
{
    protected $fillable = ['text', 'referenced_to', 'referent_id', 'replies_to'];

    protected $appends = ['rating'];


    public function getRatingAttribute()
    {
        return $this->rating = $this->votes()->where('type', '=', 'like')->count() - $this->votes()->where('type', '=', 'dislike')->count();
    }

    public function votes()
    {
        return $this->hasMany('App\Vote', 'referent_id', 'id')
            ->where('reference_to', '=', "comments");
    }

    public function parent()
    {
        return $this->hasOne($this->reference_to, 'id', 'referent_id');
    }

    public function user()
    {
        return $this->belongsTo('App\User', 'user_id', 'id');
    }
}

File: android/BettingHUB/app/src/main/java/com/bettinghub/forecasts/models/Comment.kt
package com.bettinghub.forecasts.models

import com.google.gson.annotations.SerializedName

data class Comment(
    val id: Int,
    val userId: Int,
    @SerializedName("created_at") val createdAt: String,
    var rating: Int,
    val text: String,
    @SerializedName("replies_to") val repliesTo: Int?,
    var vote: String?
) {
}
